code/modules/surgery/organs/augments_arms.dm

```diff
@@ -236,3 +236,213 @@
+	else if (implant.crystals)
+		user << "<span class='notice'>It's loaded with [implant.crystals] crystal, which [implant.charges ? "is charged":"isn't charged"].</span>"
+	else
+		user << "<span class='notice'>It's not loaded with any crystals.</span>"
+
+
+/obj/item/weapon/bluespace_crusher/proc/use_charge(mob/user, silent = FALSE)
+	if(!implant.charges)
+		if(!implant.crystals)
+			if(!silent)
+				user << "<span class='notice'>[src] requires at least a bluespace crystal to use!</span>"
+			return FALSE
+		if(!silent)
+			user << "<span class='notice'>[src] is still recharging!</span>"
+		return FALSE
+
+	implant.charges--
+	return TRUE
+
+/obj/item/weapon/bluespace_crusher/proc/attack_effect(turf/T)
+	new /obj/effect/overlay/temp/bluespace_swipe(T)
+	playsound(T,"sound/weapons/resonator_blast.ogg",50,1)
+
+
+/obj/item/weapon/bluespace_crusher/afterattack(atom/target, mob/user, proximity)
+	if(proximity)
+		if(istype(target, /obj/item) && use_charge(user))
+			attack_effect(get_turf(target))
+			new /obj/effect/overlay/temp/emp/pulse(get_turf(target))
+			user.visible_message("<span class='danger'>[user] swipes at [target] with [user.p_their()] [src], erasing it from existence!</span>", \
+			"<span class='danger'>You swipe at [target] with [src], erasing it from existence!</span>")
+			log_game("[key_name(user)] has deleted [target] with a bluespace crusher.")
+			qdel(target)
+		return
+
+	if(!use_charge(user))
+		return
+
+	attack_effect(get_step_towards(user, target))
+	new /obj/effect/overlay/temp/emp/pulse(get_turf(target))
+
+	var/list/targets = list()
+	for(var/atom/movable/A in get_turf(target))
+		if(!A.anchored)
+			targets += A
+
+	if(!isturf(target) && target in targets)
+		user.visible_message("<span class='danger'>[user] crushes spacetime, drawing [target] closer to [user.p_them()]!</span>", \
+		"<span class='danger'>You crush spacetime, drawing [target] closer to you!</span>")
+	else if(targets.len > 1)
+		user.visible_message("<span class='danger'>[user] crushes spacetime, attracting several objects!</span>", \
+		"<span class='danger'>You crush spacetime, attracting several objects!</span>")
+	else if(targets.len)
+		user.visible_message("<span class='danger'>[user] crushes spacetime, attracting [targets[1]] closer to [user.p_them()]!</span>", \
+		"<span class='danger'>You crush spacetime, drawing [targets[1]] closer to you!</span>")
+	else
+		user.visible_message("<span class='danger'>[user] crushes spacetime, but nothing happens!</span>", \
+		"<span class='danger'>You crush spacetime, but you fail to attract anything!</span>")
+
+	while(targets.len)
+		for(var/atom/movable/A in targets)
+			if(QDELETED(A) || get_dist(A,user) > 12 || A.anchored) //breaks in case of teleportation
+				targets -= A
+				continue
+			if(!A.Move(get_step_towards(A, user)))
+				targets -= A
+				continue
+			if(get_dist(A,user) <= 1)
+				targets -= A
+				continue
+		sleep(1)
```

Comment:
the delay is intentional to make the objects move smoothly, CHECK_TICK makes it instant